lib/iris/fileformats/pp.py

```diff
@@ -644,14 +644,32 @@ def ndim(self):
         return len(self.shape)
 
     def __getitem__(self, keys):
-        with open(self.path, 'rb') as pp_file:
-            pp_file.seek(self.offset, os.SEEK_SET)
-            data_bytes = pp_file.read(self.data_len)
-            data = _data_bytes_to_shaped_array(data_bytes,
-                                               self.lbpack,
-                                               self.boundary_packing,
-                                               self.shape, self.src_dtype,
-                                               self.mdi)
+        # Convert a single key to a 1-tuple, for empty-slice testing.
+        if isinstance(keys, tuple):
+            keys_tuple = keys
+        else:
+            keys_tuple = (keys,)
+
+        if any(key == slice(0, 0) for key in keys_tuple):
+            # An 'empty' slice has no data : do not open + read the file !!
+            # In these cases, return a 'fake' data array instead.
+            # Needed because, for Dask >= 2.0, "dask.array.from_array" does a
+            # fetch like [0:0, 0:0, ...], to 'snapshot' the array metadata.
+            target_shape = list(self.shape)
+            for i_dim, key in enumerate(keys_tuple):
+                if key == slice(0, 0):
+                    target_shape[i_dim] = 0
+            data = np.zeros((1,), dtype=self.dtype)
```

Comment:
Can just pass `target_shape` to `np.zeros`, without needing the extra `np.broadcast_to` step

Reply:
:+1: Good thinking. I will check on this.